File: library/src/main/java/com/alamkanak/weekview/EventClickListener.java
package com.alamkanak.weekview;

import android.graphics.RectF;
import android.support.annotation.NonNull;

public interface EventClickListener<T> {

    /**
     * Triggered when clicked on one existing event
     *
     * @param data:     event clicked.
     * @param eventRect: view containing the clicked event.
     */
    void onEventClick(@NonNull T data, @NonNull RectF eventRect);

}

File: library/src/main/java/com/alamkanak/weekview/EventClickListener.kt
package com.alamkanak.weekview

import android.graphics.RectF

@FunctionalInterface
interface EventClickListener<T> {

    /**
     * Called when an [EventChip] is clicked.
     *
     * @param data The [T] object associated with the [EventChip]'s [WeekViewEvent]
     * @param eventRect The [RectF] of the [EventChip]
     */
    fun onEventClick(data: T, eventRect: RectF)

}
